utilities/src/rand.rs

```diff
@@ -140,3 +141,39 @@ impl rand::RngCore for TestRng {
 }
 
 impl rand::CryptoRng for TestRng {}
+
+/// A cache of numbers LARPing as an RNG used **solely** for testing and benchmarking, **not** for any real world purposes.
+pub struct TestMockRng {
```

Comment:
Are there any special features of this RNG that make it inadvisable to use `TestRng`?

Reply:
Yes, we load it with predetermined "random" numbers. This was a hack to import randomly generated challenges from the Sage script.

Thinking about it again, the other direction is likely way cleaner and by now very easy to adjust to. @iamalwaysuncomfortable if you agree you could adjust the code so:
1. let the test use `TestRng::fixed` with a hardcoded seed (per set of test vectors), and we get rid of `TestMockRng`
2. run it once with `create_test_vectors` to export test vectors.
3. adjust the Sage code to read the challenges instead of [randomly generating them](https://github.com/search?q=repo%3AAleoHQ%2Fvaruna-sage-impl%20random_element&type=code)

Reply:
Tagging @ljedrz here for visibility, as he helped design the original `TestRng`.

@ljedrz we have a SAGE program that generates the randomness for test vectors and would like to unify them with the Rust test's RNG usage so the test vectors will match.

Reply:
The suggestion by @vicsn to generate test vectors using `TestRng::fixed` and to later reuse it sounds good to me :+1:; we should only make sure to comment on the fact why a fixed seed is used in those tests.

Reply:
> adjust the Sage code to read the challenges instead of [randomly generating them](https://github.com/search?q=repo%3AAleoHQ%2Fvaruna-sage-impl%20random_element&type=code)

As of a few weeks ago, the sagemath code already reads from the test vector challenges, so I'll update the rng now and just re-regenerate the test vectors.